tests/unit/students-serializer.test.js

```diff
@@ -0,0 +1,382 @@
+    const schema = {
+      links: {
+        self: `${fakeBaseUrl}/${resourceType}`,
+      },
+      data: {
+        id: fakeId,
+        type: resourceType,
+        links: { self: null },
+      },
+    };
+    if (resourceAttributes) {
+      schema.data.attributes = resourceAttributes;
+    }
+    return schema;
+  };
+
+  /**
+   * @summary Helper function for lite-testing single resource
+   * @function
+   * @param {object} serializedResource serialized resource
+   * @param {string} resourceType resource type
+   * @param {string} nestedProps field name of the nested properties
+   */
+  const testSingleResource = (serializedResource, resourceType, nestedProps) => {
+    expect(serializedResource).to.containSubset(resourceSubsetSchema(resourceType));
+
+    if (nestedProps) {
+      expect(serializedResource).to.have.nested.property(`data.attributes.${nestedProps}`);
+    }
+  };
+
+  /**
+   * @summary Helper function for lite-testing multiple resources
+   * @function
+   * @param {Object} serializedResources serialized resources
+   * @returns {Object} data object from serialized resources for further use
+   */
+  const testMultipleResources = (serializedResources) => {
+    const serializedResourcesData = serializedResources.data;
+    expect(serializedResources).to.have.keys('data', 'links');
+    expect(serializedResourcesData).to.be.an('array');
+
+    return serializedResourcesData;
+  };
+
+  /**
+   * @summary Helper function to get definition from openapi specification
+   * @function
+   * @param {string} definition the name of definition
+   * @param {Object} nestedOption nested option
+   * @param {boolean} dataItem a boolean value which represents whether it's a data item
+   * @param {string} dataField data field name
+   * @returns {Object}
+   */
+  const getDefinitionProps = (definition, nestedOption) => {
+    let result = openapi.definitions[definition].properties;
+    if (nestedOption) {
+      const { dataItem, dataField } = nestedOption;
+      if (dataItem) {
+        result = result.data.items.properties.attributes.properties;
+      } else if (dataField) {
+        result = result.data.properties.attributes.properties[dataField].items.properties;
+      }
+    }
+    return result;
+  };
+
+  /**
+   * @summary Helper function to check certain fields are parsed as numbers
+   * @function
+   * @param {Object} resource resource to be checked
+   * @param {string[]} numberFields numbers fields
+   */
+  const expectNumberFields = (resource, numberFields) => {
+    _.each(numberFields, (numberField) => {
+      expect(resource[numberField]).to.be.a('number');
+    });
+  };
+
+  it('test fourDigitToTime', () => {
+    const { fourDigitToTime } = studentsSerializer;
+    expect(fourDigitToTime(null)).to.be.null;
+
+    const invalidStrings = [];
+    while (invalidStrings.length < 10) {
+      const length = Math.floor(Math.random() * Math.floor(10));
```

Comment:
Could `Math.floor(10)` be replaced by just `10`?

Reply:
Sure. Not sure what I was thinking. Fixed.